backend/src/nodes/nodes/image_utility/convolve.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import cv2
+import numpy as np
+
+from . import category as ImageUtilityCategory
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import ImageInput, NumberInput, TextInput
+from ...properties.outputs import ImageOutput
+from ...properties import expression
+
+
+@NodeFactory.register("chainner:image:image_convolve")
+class ImageConvolveNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Convolves input image with input kernel (kernel values separated by commas)"
+        self.inputs = [
+            ImageInput("Grayscale Image", channels=1),
+            TextInput("Kernel String"),
+            NumberInput("Kernel Width/Height", minimum=0, default=3),
+            NumberInput("Padding", minimum=0, default=0),
+            NumberInput("Strides", minimum=0, default=1),
+        ]
+        self.outputs = [
+            ImageOutput(image_type=expression.Image(size_as="Input0"), channels=1)
```

Comment:
We have to account for padding and stride here to compute the correct output size. Assuming padding and stride as explained [here](https://adeshpande3.github.io/A-Beginner's-Guide-To-Understanding-Convolutional-Neural-Networks-Part-2/), the output type has to be this:

```
let w = Input0.width;
let h = Input0.height;

let kernel = Input1;

let padding = Input2;
let stride = Input3;

Image {
	width: int & floor((w + padding * 2 - 1) / stride + 1),
	height: int & floor((h + padding * 2 - 1) / stride + 1),
	channels: 1,
}
```

Sorry for the complex type expression. We have a whole custom language for [chainner's type system](https://github.com/chaiNNer-org/Navi/blob/main/navi.md) called Navi. This might be a bit overwhelming, so I'll keep it short. `let foo = value;` are "variables" (you can't reassign them), `InputX` are the types of the inputs starting with `Input0`, and the last `Image { ... }` is an image type with the specified dimensions.

If we choose to crop filter2D's output, we'll have to change the formula a bit. I'll help again then.

```suggestion
            ImageOutput(
                image_type="""
                    let w = Input0.width;
                    let h = Input0.height;

                    let kernel = Input1;

                    let padding = Input2;
                    let stride = Input3;

                    Image {
                        width: int & floor((w + padding * 2 - 1) / stride + 1),
                        height: int & floor((h + padding * 2 - 1) / stride + 1),
                        channels: 1,
                    }
                """,
                channels=1
            )
```